src/hotspot/share/gc/shenandoah/shenandoahHeap.cpp

```diff
@@ -1778,6 +1787,35 @@ void ShenandoahHeap::print_tracing_info() const {
   }
 }
 
+void ShenandoahHeap::on_cycle_start(GCCause::Cause cause, ShenandoahGeneration* generation) {
+  log_info(gc)("on_cycle_start: %s", generation->name());
```

Comment:
What is that logging for/ what does the log message mean? I'd either improve the log message or remove the logging (or make it dev+trace) if it was only for dev purposes.